### ISSUE TITLE
Build failure on RHEL 8

### ISSUE THREAD

**Issue by blackknight36** [collectd/collectd#3198]
Building collectd on RHEL 8 fails due to an error as shown below.

```
make[1]: Entering directory '/usr/src/collectd'
  CC       src/collectd_nagios-collectd-nagios.o
  CC       src/libcollectdclient/libcollectdclient_la-client.lo
  CC       src/libcollectdclient/libcollectdclient_la-network.lo
  CC       src/libcollectdclient/libcollectdclient_la-network_buffer.lo
  CC       src/libcollectdclient/libcollectdclient_la-network_parse.lo
  CC       src/libcollectdclient/libcollectdclient_la-server.lo
  CCLD     libcollectdclient.la
  CCLD     collectd-nagios
  CC       src/collectd_tg-collectd-tg.o
  CC       src/utils/heap/heap.lo
  CCLD     libheap.la
  CCLD     collectd-tg
  CC       src/collectdctl-collectdctl.o
src/collectdctl.c: In function ‘parse_identifier’:
src/collectdctl.c:169:50: error: ‘snprintf’ output may be truncated before the last format character [-Werror=format-truncation=]
     snprintf(ident_str, sizeof(ident_str), "%s/%s", hostname, value);
                                                  ^
src/collectdctl.c:169:5: note: ‘snprintf’ output 2 or more bytes (assuming 1025) into a destination of size 1024
     snprintf(ident_str, sizeof(ident_str), "%s/%s", hostname, value);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:7946: src/collectdctl-collectdctl.o] Error 1
make[1]: Leaving directory '/usr/src/collectd'
make: *** [Makefile:4844: all] Error 2
```

I've made some changes to avoid compiler errors with GCC 8.2.   Can anybody provide some feedback to make sure I'm doing this the right way?

**Comment by mrunge** [comment]
There is another proposal for this issue: https://github.com/collectd/collectd/pull/3153

**Comment by blackknight36** [comment]
@mrunge Thanks.  That appears to be a better solution.  I can run tests on RHEL 8 if needed.

**Comment by mrunge** [comment]
Again, thank you for reporting this issue.

#3153 was merged into master and also into 5.9 branch. I'm expecting another 5.9.x release soon.
